test/provider/dhkey.c

```diff
@@ -355,13 +356,32 @@ static const unsigned int required_ica_mechs[] = { RSA_ME };
 static const unsigned int required_ica_mechs_len =
                         sizeof(required_ica_mechs) / sizeof(unsigned int);
 
+typedef unsigned int (*ica_get_functionlist_t)(libica_func_list_element *,
+                                               unsigned int *);
+
 int check_libica()
 {
     unsigned int mech_len, i, k, found = 0;
     libica_func_list_element *mech_list = NULL;
+    void *ibmca_dso;
+    ica_get_functionlist_t p_ica_get_functionlist;
     int rc;
 
-    rc = ica_get_functionlist(NULL, &mech_len);
+    ibmca_dso = dlopen(LIBICA_NAME, RTLD_NOW);
+    if (ibmca_dso == NULL) {
+        fprintf(stderr, "Failed to load libica '%s'!\n", LIBICA_NAME);
+        return 77;
```

Comment:
In libica we have a constant for the automake test skip rc:
#define TEST_SKIP	77